### PR TITLE
Fix support for NRF52

### PR DESCRIPTION
**Pull request by edwardalee**
With the refactoring of platform support, code generation for NRF52 stopped working. This gets it working again, albeit with quite a few limitations.  To program an nRF52 board, we suggest starting with the [lf-nRF52 repository](https://github.com/lf-lang/lf-nRF52).  The limitations are documented there.

## Summary by CodeRabbit

- **Bug Fixes**
  - Corrected system name from "Nrf52" to "nRF52" in various configuration files for consistency.

- **Refactor**
  - Improved variable declarations and reuse in the scheduling function for better efficiency.
  - Updated interrupt handling and critical section logic to handle nesting correctly.

- **Documentation**
  - Updated platform name in documentation comments to "nRF52" for accuracy.
  
- **Enhancements**
  - Added macro `platform_define` to apply preprocessor definitions based on defined variables.

### CODE REVIEW

File: low_level_platform/api/CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@ target_include_directories(lf-low-level-platform-api INTERFACE ${CMAKE_CURRENT_L
 add_library(lf::low-level-platform-api ALIAS lf-low-level-platform-api)
 target_link_libraries(lf-low-level-platform-api INTERFACE lf::tag-api)
 
-if(${CMAKE_SYSTEM_NAME} STREQUAL "Nrf52")
+if(${CMAKE_SYSTEM_NAME} STREQUAL "nRF52")
```
Comment (lhstrh) [inline low_level_platform/api/CMakeLists.txt:6]: We've been using CamelCase for platforms, I think. If instead we want to capitalize platforms the way they are commonly referred to, we should also change `MacOS` to `macOS`. There might be more such cases...

**Comment by lhstrh** [comment]
Any particular reason this is still marked as draft?

File: low_level_platform/api/CMakeLists.txt (hunk 1 above)
Comment (edwardalee) [reply low_level_platform/api/CMakeLists.txt:6]: I'm not sure our arbitrary policy should override what the devices are called.  Also, why isn't the correct camel case NRF52?  The acronym is spelled out when you say it, which is why OS is capitalized in macOS.

Comment (lhstrh) [reply low_level_platform/api/CMakeLists.txt:6]: Good point. I'm fine with the change, honestly...

**Comment by lhstrh** [comment]
@edwardalee what should we do with this PR? Include it in `0.8.0` or not?

File: low_level_platform/impl/src/lf_nrf52_support.c [hunk 2]
```diff
@@ -258,23 +263,41 @@ int _lf_interruptable_sleep_until_locked(environment_t* env, instant_t wakeup_ti
...
- * @return int
+ * @return 0
  */
-int lf_enable_interrupts_nested() { return sd_nvic_critical_region_enter(&_lf_nested_region); }
+int lf_enable_interrupts_nested() {
+  if (_lf_nested_count == 0)
+    return 1; // Error. Interrupts have not been disabled.
+  _lf_nested_count--;
+  return sd_nvic_critical_region_exit(0);
+  // FIXME: If softdevice is not enabled, do the following instead of above:
```
Comment (lhstrh) [inline low_level_platform/impl/src/lf_nrf52_support.c:283]: Is there an issue for this? If not, we should create it, and reference it here.

**Review by lhstrh — APPROVED** [review]

File: low_level_platform/impl/src/lf_nrf52_support.c (hunk 2 above)
Comment (edwardalee) [reply low_level_platform/impl/src/lf_nrf52_support.c:283]: I have created an issue: https://github.com/lf-lang/reactor-c/issues/460

**Pull request merged**